### PR TITLE
refactor(aerorsync): own the delta engine and pin the app import budget

### PR DESCRIPTION
**Pull request by axpnet**
## Description

First tranche (A1) of Phase A of the AeroRsync standalone-crate plan, preceded by a small A0 that makes the frozen real-rsync oracle tests actually run. Three commits, each green on its own.

What changes. The `aerorsync` module now owns the delta engine: `src/delta_sync.rs` becomes `src/aerorsync/delta_engine.rs` (git mv, header intact, the inner `allow(dead_code)` dropped because it had no effect on `pub` items). `fixtures.rs` no longer imports `crate::number_parsing`: `parse_u64_loose`, `parse_f64_loose` and their three private helpers are copied verbatim as `pub(crate)` items, and `number_parsing::tests::aerorsync_fixture_parsers_stay_identical` pins the copy to the original on the rsync `--stats` shapes and the edge cases. The atomic write helpers (`WriteAtomicError`, `write_atomic_chunked`, `write_atomic_chunked_sparse`, their core, `temp_path_for` and its counter and uniqueness test) move from `delta_transport_impl.rs` to `streaming_writer.rs`, which already depended on them the wrong way round; `TEMP_SUFFIX` becomes one `pub(crate)` constant in the writer and `map_write_atomic_error` stays where `RsyncError` is. The only application consumer of the engine, the `delta_sync_analyze` Tauri command, is gated on the `aerorsync` feature like `local_sync_run`, so the default build registers the same commands and the feature-off build stays green.

Why it is byte-neutral. No wire encoding, no transfer path and no product behaviour is touched: the change is file moves, import rewrites, a verbatim copy and comment updates. The frozen real-rsync oracles are unchanged (sha256 manifest of the 19 transcript files identical before and after the suite), the temp file name formats are unchanged, the production `unsafe` count is unchanged (19, pinned by `production_unsafe_surface_matches_the_documented_count`), and the engine/adapter parity pins in `tests.rs` run without any logic change.

The import budget. `tests::app_import_budget_matches_the_documented_inventory` scans the flat module directory and pins every `crate::<application module>` occurrence to a documented inventory that may only shrink: 29 occurrences over `delta_transport`, `rsync_output` and `rsync_over_ssh` in 5 files, down from 42 lines over 5 application modules before this PR. `crate::aerorsync` is excluded as a whole identifier, so `crate::aerorsync_adapter` (the adapter the A2 tranche creates) counts; grouped imports, `super::super::`, path attributes, include macros, `macro_rules` and `.rs` files in subdirectories are refused because the scan cannot see through them. The test was seen red three times before the commit, on a plain application import, on a grouped import and on the adapter form, and green after each line was removed.

A0 and why. `artifacts_real/` was gitignored as a whole, but `fixtures.rs` reads `artifacts_real/frozen` at runtime and 42 tests (38 in `tests.rs`, 4 in `native_driver.rs`, 2 of them `#[ignore]`) return early with an `eprintln` when it is missing; `cargo test` captures the output of passing tests, so on CI and in every fresh worktree 40 oracle pins were green without reading a byte. A0.1 makes `download_signature_strong_uses_seeded_xxh128_like_frozen_oracle` derive the basis block from the harness seed formula (`i % 251`) instead of reading a second gitignored workspace file that rsync rewrites during the capture (verified: the first 700 bytes, the frozen `block_length`, equal the formula). A0.2 versions the 19-file, 92 KB transcript (captured 2026-04-17 against rsync 3.2.7, protocol 31 client and 32 server, no key and no password) with an ignore rule of `artifacts_real/*` plus a negation for `frozen/`.

Counts. AeroRsync suite (`cargo test --features aerorsync --lib aerorsync`): 677 passed, 0 failed, 15 ignored before; 697 passed, 0 failed, 15 ignored after (+18 engine tests in their new home, +1 budget test, +1 parity test selected by name). Skipped oracle pins with `--nocapture`: 40 before A0, 0 after. `cargo check --no-default-features --lib`, `cargo clippy --lib --features aerorsync --all-targets -- -D warnings`, `cargo fmt --all -- --check` and `git diff --check` all pass.

Follow-ups logged, not in this PR. Removing `delta_sync_analyze`, which has no frontend caller (with `docs/COMMAND-INVENTORY.json` and the `cli-smoke.yml` counts). Turning the oracle skip into a failure in CI. `serialize_delta`, `deserialize_delta` and `compute_signatures_streaming` have no production consumer and stay for the Phase D API decision. The engine is now reachable as `ftp_client_gui_lib::aerorsync::delta_engine`.

## Type of Change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [x] Refactor, byte-neutral: no wire encoding and no product behaviour changes

## Checklist
- [x] My code follows the project's code style
- [x] I have tested my changes
- [x] I have updated the documentation (if needed)
- [x] My changes generate no new warnings
- [x] Every commit is signed off (`git commit -s`), see [CONTRIBUTING](../CONTRIBUTING.md#sign-off-developer-certificate-of-origin)

## Related Issues
None. First tranche of the AeroRsync standalone-crate track; the release tracker is updated at merge.

## Summary by CodeRabbit

* **Reliability**
  * Improved Aerorsync transfer handling with safer atomic writes and support for sparse file data.
  * Added safeguards to prevent temporary-file naming conflicts during concurrent operations.

* **Compatibility**
  * Preserved delta-processing behavior while moving Aerorsync functionality behind its dedicated feature configuration.

* **Testing**
  * Added frozen transfer records and validation coverage for upload/download parity, file statistics, timestamps, and supported rsync versions.

* **Documentation**
  * Updated known-limitations documentation to reflect completed download memory-cap work.

### CODE REVIEW

File: src-tauri/src/aerorsync/capture/artifacts_real/frozen/download/client.stderr.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Warning: Permanently added '[127.0.0.1]:2224' (ED25519) to the list of known hosts.
```
[the remark this answers is not in the document]
Comment (axpnet) [reply src-tauri/src/aerorsync/capture/artifacts_real/frozen/download/client.stderr.txt:1]: Not a regression in the records, so I am leaving them as captured. This transcript is the lane 3 oracle, captured on 2026-04-17 against the stock rsync 3.2.7 fixture in docker-compose.real-rsync.yml, which listens on port 2224 and serves /workspace/real/ (capture/README.md, lanes table; REAL_RSYNC_LANE_PORT is pinned to 2224 in tests.rs). capture_wrapper_transcripts.sh is lane 1, the wrapper baseline on port 2222 with its own /workspace/upload and /workspace/download paths, so the two contracts differ by design.

The tests read only the four capture_in.bin and capture_out.bin files (fixtures.rs, try_load_frozen); client.stderr.txt, client.stdout.txt and remote_command.txt are the provenance of that capture and are kept verbatim so the oracle bytes can be traced to the exact command and server that produced them.

Regenerating the transcript would replace the byte-level oracle this PR must not touch; new captures belong to the multi-entry measurement step, where they get their own frozen set and provenance.
